### PR TITLE
Add qualification level to application qualifications

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

We are not saving the qualification level (if the degree is Bachelor, Master, Foundation) into the database so we can't infer what the qualification type is when the user chooses "Bachelor", "Master", "Doctorate" or "Foundation", which can lead to bugs where we display the qualification level and the type of the user degrees.

## Changes proposed in this pull request

Add a column to the database where we will map the qualification level to one of the options ("Bachelor", "Master", "Doctorate" or "Foundation").

## Link to Trello card

https://trello.com/c/dt10YnYp/55-degree-type-free-text-bug

### CODE REVIEW

File: config/analytics.yml [hunk 1]
```diff
@@ -154,6 +154,7 @@ shared:
     - predicted_grade
     - public_id
     - qualification_type
+    - qualification_level
```
Comment (cwrw) [inline config/analytics.yml:157]: could we let the data team know if this change please (if you haven't already :D)

File: spec/factories/application_qualification.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@
     application_form
     level { %w[degree gcse other].sample }
     qualification_type { %w[BA Masters A-Level gcse].sample }
+    qualification_level { %w[Bachelor Master Foundation Doctorate].sample }
```
Comment (cwrw) [inline spec/factories/application_qualification.rb:6]: I wonder if we should leave the factory till the next pr as usually we shouldn't be adding code that accesses new fields in the same pr (one on the checklist)

**Review by frankieroberto — COMMENTED** [review]
Should we also store a reference to the qualification level uuid from the [dfe-reference-data](https://github.com/DFE-Digital/dfe-reference-data) gem, like we do for the other fields now?

eg bachelors degree is given an ID here: https://github.com/DFE-Digital/dfe-reference-data/blob/main/lib/dfe/reference_data/qualifications.rb#L67

**Review by cwrw — APPROVED** [review]

File: config/analytics.yml (hunk 1 above)
Comment (tomas-stefano) [reply config/analytics.yml:157]: Yes, I told them last Friday! ✌️ 🚀

File: spec/factories/application_qualification.rb (hunk 2 above)
Comment (tomas-stefano) [reply spec/factories/application_qualification.rb:6]: Yes, agreed. I'll remove from this PR.

**Comment by tomas-stefano** [comment]
> Should we also store a reference to the qualification level uuid from the [dfe-reference-data](https://github.com/DFE-Digital/dfe-reference-data) gem, like we do for the other fields now?
> 
> eg bachelors degree is given an ID here: https://github.com/DFE-Digital/dfe-reference-data/blob/main/lib/dfe/reference_data/qualifications.rb#L67

That's an excellent idea. @cwrw can I add one more migration to this PR or to another? (and let the data team know about it)

**Comment by cwrw** [comment]
@tomas-stefano feel free to add to this

**Review by cwrw — APPROVED** [review]

**Pull request merged**